doc/nrf/libraries/others/trusted_storage.rst

```diff
@@ -0,0 +1,153 @@
+.. trusted_storage_readme:
+
+Trusted storage
+###############
+
+.. contents::
+   :local:
+   :depth: 2
+
+Overview
+********
+
+The trusted storage library enables its users to provide integrity, confidentiality and authenticity of stored data using AEAD algorithms or cryptographic hash.
+The library implements the PSA Certified Secure Storage API.
+
+Architecture
+============
```

Comment:
Sorry for the miss understanding was more thinking about the Architecture of this subsystem.


Feel free to pre-hrase this nicely.


The subsystem in designed in a modular way to allow a degree of customization. 
This is realized by using backends for the different tasks within the subsystem. 
There is the choice of the `ENCRYPTED_STORAGE_BACKEND` that responsible to modify the assets before they get forwarded to the `ENCRYPTED_STORAGE_STORAGE_BACKEND` which takes care of storing the data in Non-volatile storage. 

The provided backends are also modular to customize which crypto implementation and the key, nonce are provided.

Details about the single backends and their choices are described in -> Backends 

The figure XX gives and overview of the default architecture of the subsystem.